source/extensions/common/wasm/wasm.h

```diff
@@ -271,7 +267,7 @@ class Wasm : public Envoy::Server::Wasm,
   absl::string_view id() const { return id_; }
   WasmVm* wasmVm() const { return wasm_vm_.get(); }
   Context* generalContext() const { return general_context_.get(); }
-  Upstream::ClusterManager* clusterManager() const { return clusterManager_; }
+  Upstream::ClusterManager& clusterManager() const { return cluster_manager_; }
```

Comment:
Return non-const reference in const method? Not very reasonable to me...

Reply:
The clusterManager is a global singleton.  It is not owned by Wasm.  This is a bit confusing and a more typical/transparent style would be to store this as a pointer to make it clear that it isn't integral to the Wasm object, but this is the way that envoy does it so I am conforming to the existing style.